pkg/util/updater/updater.go

```diff
@@ -86,60 +88,64 @@ const (
 // NOTE: Update may call os.Exit.
 func (u *Updater) Update(updateConfig UpdateConfig) (updated bool, err error) {
 	if !atomic.CompareAndSwapInt32(&u.updating, 0, 1) {
-		return false, ErrAlreadyStarted
+		return false, errors.New("updating already started")
 	}
 	defer atomic.StoreInt32(&u.updating, 0)
 
-	u.status.Set("Started, checking update")
-
-	version, err := u.getVersion(updateConfig)
+	current, err := user.Current()
 	if err != nil {
-		return false, err
-	}
-
-	// No update is available.
-	if version == "" {
-		return false, nil
+		u.status.Set("failed checking permissions")
+		u.log.Error("failed checking permissions")
+		return false, errors.New("failed checking permissions")
 	}
-
-	u.status.Set(fmt.Sprintf("Found version %q", version))
-
-	u.status.Set(fmt.Sprintf("Checking/Adding repo %s", "https://deb.skywire.skycoin.com/")) // add if not exist
-	if err := u.addRepo(); err != nil {
-		return false, err
+	if current.Uid != "0" {
+		u.status.Set("insufficient permissions")
+		u.log.Error("insufficient permissions")
+		return false, errors.New("insufficient permissions")
 	}
 
-	u.status.Set("Update repositories")
-	if err := u.aptUpdate(); err != nil {
-		return false, err
+	var si sysinfo.SysInfo
+	si.GetSysInfo()
+	if (si.OS.Vendor != "debian") && (si.OS.Vendor != "arch") {
+		u.status.Set(fmt.Sprintf("updates not supported on this operating system: %s", si.OS.Vendor))
+		return false, errors.New("operating system not supported")
 	}
-	u.log.Info("Updating repositories by 'apt update' compeleted.")
 
-	// uninstall current installed skywire if its version is equal or lower that 0.4.2
-	currentVersion, err := currentVersion()
-	if err != nil {
-		return false, err
-	}
-	if currentVersion.Minor == 4 && currentVersion.Patch <= 2 {
-		u.status.Set("Uninstall current skywire version") // if needed
-		if err := u.aptRemove(); err != nil {
+	if si.OS.Vendor == "debian" {
+		u.status.Set("Checking for available package")
+		available, err := script.Exec(`apt-cache search skywire-bin`).String()
+		if err != nil {
+			return false, err
```

Comment:
I would maybe use `fmt.Errorf` here so we can add some additional context to the error. Same goes for below errors that stem from executing `apt` commands.

Reply:
sure